modules/l4regexp/matcher.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 VNXME
```

Comment:
Typically, throughout the repo, we use the same copyright (a "project" copyright as it were) and leave it to the commit history to assign credit to individual contributors.

That said, I'm not a lawyer nor do I have strong feelings about this in this case in this repo, but I could see if we end up moving this code into Caddy mainline it could be confusing with multiple copyright assignments throughout the code.

I definitely want to credit you though so what if we put a comment below the license statement and say "This file contributed by @vnxme" or something?

Reply:
Well, I just copied your typical license block and put my name instead. I think it only indicates that the file is written by me, but it doesn’t restrict any use in the project. Neither was it my intention to put any restrictions.

For example, please see how opnsense copyrights look like [here](https://github.com/opnsense/core/blob/efea5429524e847b0c1371da6c4ff666f54c2459/src/etc/rc#L3). If more than one person contributed to the file, they put multiple names of contributors sorted by years of changes.

Reply:
Oh interesting, ok. That's probably fine then.